test/yaml-rest-runner/src/main/java/org/elasticsearch/test/rest/yaml/section/PrerequisiteSection.java

```diff
@@ -0,0 +1,401 @@
+                    + "], the requires section is not properly indented"
+            );
+        }
+        String currentFieldName = null;
+        XContentParser.Token token;
+
+        while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
+            if (token == XContentParser.Token.FIELD_NAME) {
+                currentFieldName = parser.currentName();
+            } else if (token.isValue()) {
+                if ("reason".equals(currentFieldName)) {
+                    builder.setRequiresReason(parser.text());
+                } else if ("test_runner_features".equals(currentFieldName)) {
+                    parseFeatureField(parser.text(), builder);
+                } else if ("cluster_features".equals(currentFieldName)) {
+                    builder.requireClusterFeature(parser.text());
+                } else {
+                    throw new ParsingException(
+                        parser.getTokenLocation(),
+                        "field " + currentFieldName + " not supported within requires section"
+                    );
+                }
+            } else if (token == XContentParser.Token.START_ARRAY) {
+                if ("test_runner_features".equals(currentFieldName)) {
+                    while (parser.nextToken() != XContentParser.Token.END_ARRAY) {
+                        parseFeatureField(parser.text(), builder);
+                    }
+                } else if ("cluster_features".equals(currentFieldName)) {
+                    while (parser.nextToken() != XContentParser.Token.END_ARRAY) {
+                        builder.requireClusterFeature(parser.text());
+                    }
+                }
+            }
+        }
+        parser.nextToken();
+    }
+
+    public static final PrerequisiteSection EMPTY = new PrerequisiteSection();
+
+    private final List<Predicate<ClientYamlTestExecutionContext>> skipCriteriaList;
+    private final List<Predicate<ClientYamlTestExecutionContext>> requireCriteriaList;
+    private final List<String> yamlRunnerFeatures;
+    final String skipReason;
+    final String requireReason;
+
+    private PrerequisiteSection() {
+        this.skipCriteriaList = new ArrayList<>();
+        this.requireCriteriaList = new ArrayList<>();
+        this.yamlRunnerFeatures = new ArrayList<>();
+        this.skipReason = null;
+        this.requireReason = null;
+    }
+
+    PrerequisiteSection(
+        List<Predicate<ClientYamlTestExecutionContext>> skipCriteriaList,
+        String skipReason,
+        List<Predicate<ClientYamlTestExecutionContext>> requireCriteriaList,
+        String requireReason,
+        List<String> yamlRunnerFeatures
+    ) {
+        this.skipCriteriaList = skipCriteriaList;
+        this.requireCriteriaList = requireCriteriaList;
+        this.yamlRunnerFeatures = yamlRunnerFeatures;
+        this.skipReason = skipReason;
+        this.requireReason = requireReason;
+    }
+
+    public boolean yamlRunnerHasFeature(String feature) {
```

Comment:
nit, maybe `yamlRunnerRequiresFeature` is easier to understand?

Reply:
For consistency with `ClientYamlTestSuite`, I have renamed all as `hasYamlRunnerFeature`